brian2/core/variables.py

```diff
@@ -761,14 +757,14 @@ def __init__(self, name, variable, group, unit=None):
         # We keep a strong reference to the `Indexing` object so that basic
         # indexing is still possible, even if the group no longer exists
         self.indexing = self.group._indices
-        self.unit = unit
+        self.dim = dimensions
```

Comment:
I'm actually slightly concerned that we have dim and ndim now that I think about it, but I think it's probably OK.